src/providers/ogr/qgsogrconnpool.h

```diff
@@ -61,7 +61,13 @@ class QgsOgrConnPoolGroup : public QObject, public QgsConnectionPoolGroup<QgsOgr
     Q_OBJECT
 
   public:
-    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ) { initTimer( this ); }
+    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ), mRefCount( 0 ) { initTimer( this ); }
+    void ref() { ++mRefCount; }
```

Comment:
Possible scenario with > 1: loaded layer which is being rendered, and user created a feature source manually with `QgsVectorDataProvider::featureSource`. And also the scenario where the same data source has been added multiple times.